examples/10_recipe.py

```diff
@@ -0,0 +1,451 @@
+
+- When we add a processing step, we can specify a range for any of its
+  parameters, rather than a single value. Thus the ranges of hyperparameters to
+  consider for tuning are provided directly inside the estimator that uses
+  them. Moreover, these hyperparameters can be given human-readable names which
+  are useful for inspecting hyperparameter search results.
+
+.. admonition:: The Recipe has no ``fit()`` or ``predict()``
+
+    The recipe is not a scikit-learn estimator with ``fit`` and ``predict``
+    methods. Rather, it is a tool for configuring a scikit-learn estimator.
+    Once we have created a Recipe, we must call one of its methods such as
+    ``get_pipeline``, ``get_grid_search``, or ``get_randomized_search`` to
+    obtain the corresponding scikit-learn object.
+
+    For example:
+
+    - ``get_pipeline`` returns a scikit-learn |Pipeline| which applies all the
+      steps we have configured without any hyperparameter tuning.
+    - ``get_grid_search`` returns a scikit-learn |GridSearchCV| which uses the same
+      pipeline but runs a nested cross-validation loop to evaluate different
+      combinations of hyperparameters from the ranges we have specified and select
+      the best configuration.
+
+.. |Pipeline| replace::
+    :class:`~sklearn.pipeline.Pipeline`
+
+.. |GridSearchCV| replace::
+    :class:`~sklearn.model_selection.GridSearchCV`
+
+.. |RandomizedSearchCV| replace::
+    :class:`~sklearn.model_selection.GridSearchCV`
+
+.. |HGB| replace::
+    :class:`~sklearn.ensemble.HistGradientBoostingRegressor`
+
+.. |TargetEncoder| replace::
+    :class:`~sklearn.preprocessing.TargetEncoder`
+
+.. |Recipe| replace::
+     :class:`~skrub.Recipe`
+
+.. |TableReport| replace::
+     :class:`~skrub.TableReport`
+
+.. |DatetimeEncoder| replace::
+     :class:`~skrub.DatetimeEncoder`
+
+.. |MinHashEncoder| replace::
+     :class:`~skrub.MinHashEncoder`
+
+"""
+
+
+# %%
+# Getting a preview of the transformed data
+# -----------------------------------------
+#
+# The |Recipe| can be initialized with the full dataset (including the
+# prediction target, ``y``). We can tell it which columns constitute the target
+# and should be kept separate from the rest.
+
+# %%
+from skrub import Recipe, datasets
+
+dataset = datasets.fetch_employee_salaries()
+df = dataset.X
+df["salary"] = dataset.y
+
+recipe = Recipe(df, y_cols="salary", n_jobs=8)
+recipe
+
+# %%
+# Our recipe does not contain any transformations yet, except for
+# the built-in one which separates the target columns from the predictive
+# features.
+
+# %%
+# We can use ``sample()`` to get a sample of the data transformed
+# by the steps we have added to the recipe so far. The recipe draws a
+# random sample from the dataframe we used to initialize it, and applies the
+# transformations we have specified.
+#
+# At this point we have not added any transformations yet, so we just get a
+# sample from the original dataframe.
+
+# %%
+recipe.sample()
```

Comment:
yes my goal is not to get the display, but to show that you can get the actual data in a dataframe (and not only a display). maybe instead of outputing the sample I should print its shape or type or something like that

Reply:
Right, I understand your point now. Yes, I think that using `.shape`  would make it easier to understand. Maybe mentioning `get_X` and `get_y` as well?